src/probnum/_config.py

```diff
@@ -0,0 +1,83 @@
+import contextlib
+from typing import Any
+
+
+class Configuration:
+    """
+    Configuration over which some mechanics of probnum can be controlled dynamically.
+
+    ``probnum`` provides some configurations together with default values. These
+    are listed in the tables below.
+    Additionally, the user can register own configuration entries via the method
+    :meth:`register`. Configuration entries can only be registered once and can only
```

Comment:
How public do we want to make this functionality?

Reply:
As in, it obviously exists, but should it appear in the docs like this?

Reply:
I was asking myself the same question. I think there are arguments pro and contra, making it public.
This way the user can utilize this for their own code when using probnum, and I think they cannot break too much.

Reply:
Hmmm if it is impossible/hard to break something, I think there is nothing wrong with it. Let's see what the others think.